### PR TITLE
Reorder/Modularize networking e2e + pod launch phase, clean up comments.  

### PR DESCRIPTION
**Pull request by jayunit100**
Creating a new PR, the other seemed to die on me.

### CODE REVIEW

**Comment by jayunit100** [comment]
fixing collision now with refactored `Provider` function.

**Comment by jayunit100** [comment]
ok @satnam6502  and  @bgrant0607 ....  finally here is theupdated pr.  i killed the old one as it got broken by a conflicted commits + a messed up rebase on my end.  this is the updated pr.  Thanks !

**Comment by satnam6502** [comment]
Re-assigning for a quicker review :-)

**Comment by satnam6502** [comment]
@piosz  feel free to bounce this back to me and I shall try to get to it on Thursday.

File: test/e2e/networking.go [hunk 1]
```diff
@@ -29,6 +29,60 @@ import (
...
+	if err != nil {
+		Failf("Unable to connect/talk to the internet: %v", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		Failf("Unexpected error code, expected 200, got, %v (%v)", resp.StatusCode, resp)
+	}
+})
+
+func LaunchNetTestPodPerNode(nodes *api.NodeList, name string, c *client.Client, ns string) []string {
+
```
Comment (piosz) [inline test/e2e/networking.go]: Please add a By statement here. Also remove this empty line.

File: test/e2e/networking.go [hunk 2]
```diff
@@ -29,6 +29,60 @@ import (
...
+	podNames := []string{}
+
+	totalPods := len(nodes.Items)
+
+	Expect(totalPods).NotTo(Equal(0))
+
+	for i, node := range nodes.Items {
+		podName := fmt.Sprintf("%s-%d", name, i)
+		podNames = append(podNames, podName)
+		Logf("Creating pod %s on node %s", podName, node.Name)
```
Comment (piosz) [inline test/e2e/networking.go]: Do we really want to have this log message?

File: test/e2e/networking.go [hunk 3]
```diff
@@ -29,6 +29,60 @@ import (
...
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "webserver",
+						Image: "gcr.io/google_containers/nettest:1.1",
+						Args: []string{
+							"-service=" + name,
+							//peers >= totalPods should be asserted by the container.
+							fmt.Sprintf("-peers=%d", totalPods),
```
Comment (piosz) [inline test/e2e/networking.go]: What does mean -peers arg? Shouldn't be totalPods-1?

File: test/e2e/networking.go [hunk 4]
```diff
@@ -38,37 +92,38 @@ var _ = Describe("Networking", func() {
...
+	})
+
 	// Create a unique namespace for this test.
 	ns := "nettest-" + randomSuffix()
+	name := "nettest"
 
-	It("should function for pods", func() {
+	It("should function for intra-pod communication", func() {
 		if testContext.Provider == "vagrant" {
 			By("Skipping test which is broken for vagrant (See https://github.com/GoogleCloudPlatform/kubernetes/issues/3580)")
```
Comment (piosz) [inline test/e2e/networking.go]: #3580 only

File: test/e2e/networking.go [hunk 5]
```diff
@@ -141,7 +175,7 @@ var _ = Describe("Networking", func() {
 			}
 		}()
 
-		By("Wait for the webserver pods to be ready")
+		By("Waiting for the webserver pods to transition to Running state")
```
Comment (piosz) [inline test/e2e/networking.go]: How about moving this also to LaunchNetTestPodPerNode()?

File: test/e2e/networking.go [hunk 6]
```diff
@@ -150,7 +184,7 @@ var _ = Describe("Networking", func() {
 		By("Waiting for connectivity to be verified")
 		const maxAttempts = 60
 		passed := false
-		var body []byte
+		var body []byte //once response OK, we evaluate this variable for test pass/fail.
```
Comment (piosz) [inline test/e2e/networking.go]: Move to the previous line. I don't like having both comment and code in the same line.

File: test/e2e/networking.go [hunk 7]
```diff
@@ -29,6 +29,60 @@ import (
...
+	resp, err := http.Get("http://google.com")
+	if err != nil {
+		Failf("Unable to connect/talk to the internet: %v", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		Failf("Unexpected error code, expected 200, got, %v (%v)", resp.StatusCode, resp)
+	}
+})
+
+func LaunchNetTestPodPerNode(nodes *api.NodeList, name string, c *client.Client, ns string) []string {
```
Comment (piosz) [inline test/e2e/networking.go]: Maybe move this function to the test/e2e/util.go? It's useful function which starts pod on each node. As a parameter use docker image name.

File: test/e2e/networking.go (hunk 4 above)
Comment (jayunit100) [inline test/e2e/networking.go]: yup this should be skipped since its below the return if Provider==vagrant.

File: test/e2e/networking.go (hunk 5 above)
Comment (jayunit100) [inline test/e2e/networking.go]: since its a test, i chose to keep it in this block - and use that function as a utility instead... is that ok?

**Comment by jayunit100** [comment]
Okay, i implemented the comments inline.  Regarding "peers" its not x-1, its x.  The reason is that the service endpoints array actually returns all endpoints, so each node is a peer of itself, by definition.

File: test/e2e/networking.go (hunk 2 above)
Comment (satnam6502) [inline test/e2e/networking.go]: I find these log messages very useful for debugging.

File: test/e2e/networking.go (hunk 3 above)
Comment (satnam6502) [inline test/e2e/networking.go]: This identifies the number of peers including itself. Much of this code has already been reviewed so it's a shame that much of it is turning up in green again.

File: test/e2e/networking.go (hunk 6 above)
Comment (satnam6502) [inline test/e2e/networking.go]: I think it's perfectly fine style to have code and comment on the same line?

**Comment by jayunit100** [comment]
yup, tests pass :) rebasing and pushing.  thanks for the  fast review.

**Comment by satnam6502** [comment]
Yes, thanks to @piosz for letting us exploit a time zone difference to our advantage :-)

**Comment by piosz** [comment]
LGTM. Thanks for the change. @satnam6502 could you please do merge since I won't be online anymore?

**Pull request merged**